### PR TITLE
Add models skeleton

### PR DESCRIPTION
**Pull request by gmelodie**
Add general database schematic

- **House**
    - Invite links
    - Name
    - Description
- **UserHouse**
    - User ID
    - House ID
- **Task**
    - Name
    - Description
    - House ID
    - Frequency
- **UserTask**
    - Task ID
    - User ID
    - Deadline
    - Done

### CODE REVIEW

File: app/models.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+    def set_password(self, password) -> None:
+        self.password_hash = generate_password_hash(password)
+
+    def check_password(self, password) -> bool:
+        return check_password_hash(self.password_hash, password)
+
+class House(db.Model):
+    id = db.Column(db.Integer, primary_key=True)
+    invite_link = db.Column(db.String(64), index=True, unique=True)
```
Comment (teymour-aldridge) [inline app/models.py:20]: I think we can generate this using a JSON Web Token – basically we would encrypt a message from the server, stick it in a link and then decrypt it later.

File: app/models.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    name = db.Column(db.String(64), index=True, unique=True)
+    description= db.Column(db.String(1024))
+
+    def update_invite_link(self, link) -> None:
```
Comment (teymour-aldridge) [inline app/models.py:24]: I don't think this is so useful, because `<house instance>.invite_link = link` is just as good?

File: app/models.py [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    id = db.Column(db.Integer, primary_key=True)
+    name = db.Column(db.String(64))
+    house_id = db.Column(db.Integer)
+    description= db.Column(db.String(1024))
+    frequency = db.Column(db.Integer)
+
+class UserTask(db.Model):
+    task_id = db.Column(db.Integer, primary_key=True)
+    user_id = db.Column(db.Integer, primary_key=True)
+    # Deadline
```
Comment (teymour-aldridge) [inline app/models.py:41]:
```suggestion
    deadline = db.Column(db.DateTime)
```

File: app/models.py [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+    # done
```
Comment (teymour-aldridge) [inline app/models.py:42]:
```suggestion
    done = db.Column(db.Boolean)
```

File: app/models.py [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+        self.password_hash = generate_password_hash(password)
+
+    def check_password(self, password) -> bool:
+        return check_password_hash(self.password_hash, password)
+
+class House(db.Model):
+    id = db.Column(db.Integer, primary_key=True)
+    invite_link = db.Column(db.String(64), index=True, unique=True)
+    name = db.Column(db.String(64), index=True, unique=True)
+    description= db.Column(db.String(1024))
```
Comment (teymour-aldridge) [inline app/models.py:22]:
```suggestion
    description = db.Column(db.String(1024))
```

**Comment by teymour-aldridge** [comment]
Otherwise looks good!

File: app/models.py (hunk 1 above)
Comment (teymour-aldridge) [reply app/models.py:20]:
```suggestion
```

Comment (gmelodie) [reply app/models.py]: But wouldn't be nice to have a link that users can share between them? In this case we'd have to store it in the database

Comment (teymour-aldridge) [reply app/models.py]: I think we can generate them cryptographically!

**Pull request merged**